packages/react/src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -97,6 +99,26 @@ const ButtonGroup = ({ buttons, enableSizeByContent }) => {
     };
   }, []);
 
+  useEffect(() => {
+    setSameHeight();
+    root.addEventListener('resize', setSameHeight);
+    return () => root.removeEventListener('resize', setSameHeight);
+  }, []);
+
+  /**
+   * Set the buttons to have the same height based on the tallest one
+   */
+  const setSameHeight = () => {
+    root.requestAnimationFrame(() => {
+      const { current: containerNode } = groupRef;
+      if (containerNode) {
+        sameHeight(
```

Comment:
@caiobozato @annawen1 wonder if it's worth adding logic to only do this when there are 2 or more buttons?

Reply:
hmm good point we probably want to skip over all that logic if we're only using one button

Reply:
@annawen1 @jeffchew Conditional logic added in latest commit